clarifai.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strings"
+)
+
+const (
+	clarifaiApi = "https://api.clarifai.com/v2/models/aaa03c23b3724a16a56b629203edc62c/outputs"
+	clarifyAuth = "https://api.clarifai.com/v1/token"
+	minValue    = 0.72
+	maxTags     = 10
+)
+
+type ClarifaiClient struct {
+	Token  string
+	Expiry int
+	Access bool
+}
+
+type ClarifyTag struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+func NewClarifaiClient() *ClarifaiClient {
+	return &ClarifaiClient{
+		Access: false,
+	}
+}
+
+// Check whether the client still has access rights to Clarifai
+func (client *ClarifaiClient) isAccess() bool {
+	if !client.Access || client.Expiry <= 0 {
+		client.Access = false
+		return false
+	}
+	return true
+}
+
+// Refresh the access token
+func (client *ClarifaiClient) RefreshAccesToken() {
+	CLARIFAI_ID := os.Getenv("CLARIFAI_CLIENT_ID")
+	CLARIFAI_SECRET_KEY := os.Getenv("CLARIFAI_SECRET_KEY")
+
+	responseData := authRequest(CLARIFAI_ID, CLARIFAI_SECRET_KEY)
+	var token struct {
```

Comment:
Is this a short-hand for `type Token struct {...}` and instantiate Token? Just wondering what's the difference and when should we use it